preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/FormattingForAbstractModel.groovy

```diff
@@ -322,6 +322,43 @@ IProcess formatHydroLayer() {
     )
 }
 
+/**
+ * This process is used to transform the raw impervious table into a table that matches the constraints
+ * of the geoClimate Input Model
+ * @param datasource A connexion to a DB containing the raw impervious table
+ * @param inputTableName The name of the raw impervious table in the DB
+ * @return outputTableName The name of the final impervious table
+ */
+IProcess formatImperviousLayer() {
+    return create({
+        title "Format the raw impervious table into a table that matches the constraints of the GeoClimate Input Model"
+        inputs datasource    : JdbcDataSource, inputTableName: String, epsg: int
+        outputs outputTableName: String
+        run { datasource, inputTableName,epsg ->
+            logger.info('Impervious transformation starts')
+            outputTableName = "INPUT_IMPERVIOUS_${UUID.randomUUID().toString().replaceAll("-", "_")}"
+            datasource.execute """Drop table if exists $outputTableName;
+                        CREATE TABLE $outputTableName (THE_GEOM GEOMETRY(GEOMETRY, $epsg), id_impervious serial, ID_SOURCE VARCHAR);"""
+            logger.info(inputTableName)
+            if(inputTableName!=null){
+                datasource.withBatch(1000) { stmt ->
+                    datasource.getTable(inputTableName).eachRow { row ->
+                        def toAdd = true
+                        if ((row.surface != null) && (row.surface == "grass")) {toAdd = false}
+                        if ((row.parking != null) && (row.parking == "underground"))  {toAdd = false}
```

Comment:
What about is we use the same approach as getType to find the type associated to a set of tags.
Then, if the type is  null the values are not added.

Reply:
It's not exactly the same : there's no type to provide.
It's more like the special treatment for subways when formatting the rails layer.

Reply:
ok